tests/plugins/test_solvers.py

```diff
@@ -62,6 +70,13 @@ def test_get_cached_solver_backend(plugin_manager, mocker):
     assert mocked.call_count == 1  # real caching!
 
 
+def test_solver_user_agent(plugin_manager, monkeypatch):
+    plugin = VerboseSolverPlugin()
+    plugin_manager.register(plugin)
+    monkeypatch.setattr(context, "solver", "verbose-classic")
+    assert verbose_user_agent in context.user_agent
```

Comment:
`user_agent` is memoized, so it might be returning the old value before you patched `solver` :/

Reply:
okay, lemme work around that..

Reply:
done in https://github.com/conda/conda/pull/12104/commits/db50fde447b597da4ab631382002a98b22abf90f